### ISSUE TITLE
[Bug]: npx cap sync hooks (`capacitor:sync:before` and `capacitor:sync:after`) have changed behaviour and are breaking the build process

### ISSUE THREAD

**Issue by gabides** [ionic-team/capacitor#7581]
### Capacitor Version

6.1.0 and 6.1.1

### Platforms Affected

- [X] iOS
- [X] Android
- [ ] Web

### Current Behavior

when running `npx cap sync`, the `capacitor:sync:before` and `capacitor:sync:after` from the main package is executed for each plugin. This is making our build throw a lot of errors as we use these hooks to fix some plugins code from our capacitor project root. (it tries to execute the script  dozen of time, once for each plugin, and it fails each time as the path of execution is not correct)

### Expected Behavior

in 6.0 and before, the `capacitor:sync:before` and `capacitor:sync:after` were executed only once at the capacitor project level and not at the plugin level. 

### Project Reproduction

N/A

### Additional Information

This change looks intended, from this [PR](https://github.com/ionic-team/capacitor/pull/7499) but it is not documented. Is it expected and are those hooks not supposed to be usable like before (only once, before and after the `npx cap sync` command)?

**Comment by adrenaline15** [comment]
i do run into the same problem. downgrading to ~6.0.0 helps for now.

might be worth noting that we're using it in a nx-monorepo

**Comment by jcesarmobile** [comment]
please, can you create a separate issue and provide a sample app that reproduces your problem when using nx-monorepo?

It might be a different problem.

**Comment by jcesarmobile** [comment]
yes, it's a bug, please report it and provide a sample app.
